### DISCUSSION TITLE
How to get VS Code's "Go to definition" to work on auto-imported components?

### DISCUSSION THREAD

**Discussion by jf908** [nuxt/nuxt#17047] [Questions] [answered]
Currently using "Go to definition" on an auto imported components opens up `.nuxt\components.d.ts` instead of going to the component source file, is there any way to fix this?

Is this considered an issue with Volar or perhaps it should be solved with a Nuxt extension?

**Comment by ojvribeiro** [comment]
That's one thing on Nuxt that makes me sad, although I love Nuxt. I hope they figure it out how to do a proper go to component definition in the future. Unfortunately, the Go to Definition VSCode extension by @antfu do not seem to work most of the times, and when it works it leaves the `components.d.ts` file open.

**Reply by tilenpirih** [reply]
have the same problem and I hate it... anyone solve this problem leaving the components.d.ts open?

**Comment by viracoding** [comment]
any updates here?

**Comment by cpreston321** [comment, marked as answer]
@jf908 @ojvribeiro @viracoding 

Hello all please download this vscode ext https://marketplace.visualstudio.com/items?itemName=antfu.goto-alias. 

This will help you navigate auto imports. If this helped please marked this as answered! 

Thanks,
CP

**Reply by viracoding** [reply]
Looks good in the first tries. I will use and test it these days.
As @ojvribeiro already mentioned `components.d.ts` remains open.

**Reply by viracoding** [reply]
Go to definition does not open always the right component.

**Reply by muffincode** [reply]
works fine with volar takeover mode, thanks

**Reply by ojvribeiro** [reply]
It is now working fine for me as well, with Takeover Mode and no components.d.ts remaining open. :)

**Reply by ilyasturki** [reply]
How do you enable takeover mode ?

**Reply by AloisSeckar** [reply]
Takeover mode has been made obsolete with the release of [Vue official](https://marketplace.visualstudio.com/items?itemName=Vue.volar) few weeks ago. Shouldn't be needed anymore.

**Reply by Tenrys** [reply]
I feel like this should be pinned, documented or something, because this makes DX quite nicer.

**Reply by Random-nobody** [reply]
@AloisSeckar I'm using new Vue official nad `components.d.ts` still remain open. Since takeover mode has been made obsolete, is there a way to make it work correctly?

And I second @Tenrys that it should be documented somewhere. DX for someone new to Nuxt3 is really terrible at the moment.

**Reply by AAAboutajedyne** [reply]
Any updates here, I'm using Vue official and `component.d.ts` still remain open!
Another problem: when I do `go back`, it redirects me to `component.d.ts` first, so I need to do `go back` **twice** to be able to go back !

**Reply by robinbeatty** [reply]
I feel like I've seen a config in vs code to set it to automatically go to 1st definition of an alias (I.e. the one defined in compenents.d.ts) which is what we want right? Can't for the life of me find it though. Anyone else seen this config?

**Reply by Tenrys** [reply]
That would be

```json
{
  "editor.gotoLocation.multipleDefinitions": "goto"
}
```

It's actually indicated in the goto alias plugin's README, although I don't think it used to be there up until recently.

**Reply by immx2** [reply]
This doesn't seem to be working anymore? Nuxt 4.4.7

**Comment by ptoussai** [comment]
"Go to Source Definition" works for me out of the box.

**Reply by nawazishali** [reply]
What version on nuxt and vscode do you use? It still doesn't work for me.

**Comment by rasibn** [comment]
Unfortune but this issue makes nuxt painful to work with compared to its alternatives.

**Reply by robinbeatty** [reply]
Maybe bit of an unfair comparison since auto-imports in any other framework would face the same issue in IDE?

**Reply by rasibn** [reply]
Well, svelte doesn't require import on runes. Although they don't have auto import on components but ig I never felt like I needed that while using $lib was always easy enough?
